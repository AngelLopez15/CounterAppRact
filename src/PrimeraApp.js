import React from 'react'
// importando PropTypes 
import PropTypes from 'prop-types'

// esto es una functional components
// desestructurando la propiedad (properties) y asignandole un
// valor por default
const PrimeraApp = ({saludo, subtitulo}) =>{
    
    return (
        <>
            <h1>{saludo}</h1>
            <p>{subtitulo}</p>
        </>
    )
}

// aplicando los proptypes
PrimeraApp.propTypes ={
    // poniendo tipo de datos a la propiedad
    saludo: PropTypes.string.isRequired
    // si no se pone isRequiered entonces estamos
    // diciendo que la propiedad es opcional
    // con isRequiered le ordenamos que es obligatorio
}

// poniendo un DefaultProp para en este caso 
// un subtitulo
PrimeraApp.defaultProps={
    subtitulo: 'Soy un subtitulo'
}

export default PrimeraApp 