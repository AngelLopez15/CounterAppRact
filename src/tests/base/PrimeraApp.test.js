import React from 'react';
import {shallow} from 'enzyme'
import PrimeraApp from '../../PrimeraApp'
import '@testing-library/jest-dom'

describe('Pruebas en <PrimeraApp />', () => {
    test('debe de mostar <PrimeraApp /> correctamente', () => {
        const saludo='Hola Mundo desde un const'
        const wrapper = shallow (<PrimeraApp saludo={saludo} />)
        
        expect(wrapper).toMatchSnapshot()
    })

    test('debe del mostrar el subtitulo envia por los Props', () => {
        const saludo='Hola Mundo desde un const'
        const subtitulo='Aqui debe ir un subtitulo'
        const wrapper = shallow (
            <PrimeraApp 
                saludo={saludo} 
                subtitulo={subtitulo}
            />
        )
        const textoParrafo = wrapper.find('p').text()
        expect(textoParrafo).toBe(subtitulo)
    })
    
})
