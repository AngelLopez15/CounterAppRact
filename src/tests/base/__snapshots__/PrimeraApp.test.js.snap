// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <PrimeraApp /> debe de mostar <PrimeraApp /> correctamente 1`] = `
<Fragment>
  <h1>
    Hola Mundo desde un const
  </h1>
  <p>
    Soy un subtitulo
  </p>
</Fragment>
`;
